test/e2e/network/ingress_scale.go

```diff
@@ -0,0 +1,329 @@
+
+			By(fmt.Sprintf("Ensuring TLS secret %s...", scaleTestSecretName))
+			jig.PrepareTLSSecret(ns, scaleTestSecretName, scaleTestHostname)
+
+			currentNum := new(int)
+
+			prepareIngsFunc := func(goalNum int) {
+				var ingWg sync.WaitGroup
+				numToCreate := goalNum - *currentNum
+				ingWg.Add(numToCreate)
+				errQueue := make(chan error, numToCreate)
+				start := time.Now()
+				for ; *currentNum < goalNum; *currentNum++ {
+					suffix := fmt.Sprintf("%d", *currentNum)
+					go func() {
+						defer GinkgoRecover()
+						defer ingWg.Done()
+						acquireThread()
+						defer releaseThread()
+
+						start := time.Now()
+						svcCreated, ingCreated, err := createScaleTestServiceIngress(f.ClientSet, ns, suffix)
+						scaleTestSvcs = append(scaleTestSvcs, svcCreated)
+						scaleTestIngs = append(scaleTestIngs, ingCreated)
+						if err != nil {
+							framework.Logf("Failed to create test service and ingress: %v", err)
+							errQueue <- err
+							return
+						}
+						framework.Logf("Waiting for ingress %s to come up...", ingCreated.Name)
+						jig.WaitForGivenIngressWithTimeout(ingCreated, false, ingressCreateLatencyThreshhold)
+						elapsed := time.Since(start)
+						framework.Logf("Spent %s for ingress %s to come up", elapsed, ingCreated.Name)
+					}()
+				}
+
+				ingWg.Wait()
+				Expect(len(errQueue)).To(Equal(0), "Expect no error for creating services and ingresses")
+				elapsed := time.Since(start)
+				framework.Logf("Spent %s for %d ingresses to come up", elapsed, numToCreate)
+			}
+
+			measureCreatUpdateFunc := func() {
+				framework.Logf("Create one more ingress and wait for it to come up")
+				start := time.Now()
+				svcCreated, ingCreated, err := createScaleTestServiceIngress(f.ClientSet, ns, fmt.Sprintf("%d", *currentNum))
+				*currentNum = *currentNum + 1
+				scaleTestSvcs = append(scaleTestSvcs, svcCreated)
+				scaleTestIngs = append(scaleTestIngs, ingCreated)
+				framework.ExpectNoError(err)
+				framework.Logf("Waiting for ingress %s to come up...", ingCreated.Name)
+				jig.WaitForGivenIngressWithTimeout(ingCreated, false, ingressCreateLatencyThreshhold)
+				elapsed := time.Since(start)
+				framework.Logf("Spent %s for ingress %s to come up", elapsed, ingCreated.Name)
+
+				framework.Logf("Updating ingress and wait for change to take effect")
+				ingToUpdate, err := f.ClientSet.ExtensionsV1beta1().Ingresses(ns).Get(ingCreated.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err)
+				addTestPathToIngress(ingToUpdate)
+				start = time.Now()
+				ingToUpdate, err = f.ClientSet.ExtensionsV1beta1().Ingresses(ns).Update(ingToUpdate)
+				framework.ExpectNoError(err)
+				jig.WaitForGivenIngressWithTimeout(ingToUpdate, false, ingressUpdateLatencyThreshhold)
+				elapsed = time.Since(start)
+				framework.Logf("Spent %s for updating ingress %s", elapsed, ingToUpdate.Name)
+			}
+
+			By(fmt.Sprintf("Create small amount (%d) ingresses and wait for them to come up", numIngressesSmall))
```

Comment:
I think it would be cleaner if you put this block into a for loop and iterated against a slice. Something like:

for _, num := range [3]int{numIngressesSmall, numIngressesMedium, ...}